File: experiments/adrianco/experiment-14/runs/language=elixir_model=opus-4.8-fast_prompt=neutral/rep1/test/support/fixtures.ex
defmodule BrSoccer.Fixtures do
  @moduledoc "Small, deterministic dataset used by unit tests."

  alias BrSoccer.{Match, Player, TeamName}

  @doc "Build a match struct from raw team names (keys are derived automatically)."
  def match(competition, season, home, away, hg, ag, opts \\ []) do
    %Match{
      competition: competition,
      source: Keyword.get(opts, :source, :brasileirao_csv),
      season: season,
      date: Keyword.get(opts, :date),
      round: Keyword.get(opts, :round),
      stage: Keyword.get(opts, :stage),
      home: TeamName.display(home),
      away: TeamName.display(away),
      home_key: TeamName.key(home),
      away_key: TeamName.key(away),
      home_goal: hg,
      away_goal: ag
    }
  end

  @doc "A three-team round-robin Brasileirão 2020 plus a couple of other competitions."
  def matches do
    [
      # Brasileirão 2020 — Alpha, Beta, Gamma (single round-robin home & away)
      match(:brasileirao, 2020, "Alpha", "Beta", 2, 0, date: ~D[2020-08-01], round: 1),
      match(:brasileirao, 2020, "Beta", "Alpha", 1, 1, date: ~D[2020-08-08], round: 2),
      match(:brasileirao, 2020, "Alpha", "Gamma", 3, 1, date: ~D[2020-08-15], round: 3),
      match(:brasileirao, 2020, "Gamma", "Alpha", 0, 0, date: ~D[2020-08-22], round: 4),
      match(:brasileirao, 2020, "Beta", "Gamma", 2, 2, date: ~D[2020-08-29], round: 5),
      match(:brasileirao, 2020, "Gamma", "Beta", 1, 0, date: ~D[2020-09-05], round: 6),
      # Another season for the same clubs (Alpha vs Beta)
      match(:brasileirao, 2019, "Alpha", "Beta", 4, 2, date: ~D[2019-05-10], round: 1),
      # A cup tie and a Libertadores tie
      match(:copa_do_brasil, 2020, "Alpha", "Gamma", 1, 0, date: ~D[2020-06-01], stage: "final"),
      match(:libertadores, 2020, "Alpha", "Beta", 5, 0, date: ~D[2020-03-01], stage: "group stage")
    ]
  end

  @doc "A handful of FIFA-style players. Some clubs match Brazilian team keys."
  def players do
    [
      player(1, "Neymar Jr", "Brazil", 92, "Paris Saint-Germain", "LW", 27),
      player(2, "Alpha Star", "Brazil", 80, "Alpha", "ST", 24),
      player(3, "Beta Keeper", "Brazil", 75, "Beta", "GK", 30),
      player(4, "Foreign Guy", "Argentina", 85, "Alpha", "CB", 28),
      player(5, "Old Timer", "Brazil", 70, "Alpha", "CM", 35)
    ]
  end

  def data, do: %{matches: matches(), players: players()}

  defp player(id, name, nat, overall, club, pos, age) do
    %Player{
      id: id,
      name: name,
      nationality: nat,
      overall: overall,
      potential: overall + 2,
      club: club,
      club_key: TeamName.key(club),
      position: pos,
      age: age
    }
  end
end

File: experiments/adrianco/experiment-5/runs/language=clojure_model=claude-opus-4-8_tooling=none/rep1/test/soccer/fixtures.clj
(ns soccer.fixtures
  "=============================================================================
   soccer.fixtures — Small hand-built dataset for fast, deterministic tests
   -----------------------------------------------------------------------------
   The BDD scenarios in the *_test namespaces run against this fixture rather
   than the full CSV files so they are fast and assertion values are known
   exactly. The fixture mirrors the unified shapes produced by soccer.data.
   ============================================================================="
  (:require [soccer.normalize :as n]))

(defn- m [competition season round date home away hg ag]
  {:competition competition :season season :round round :date (n/norm-date date)
   :stage nil
   :home (n/display-name home) :away (n/display-name away)
   :home-key (n/team-key home) :away-key (n/team-key away)
   :home-goal hg :away-goal ag :stats nil :source "fixture"})

(def matches
  [;; Brasileirão 2019 — a tiny 3-team round robin (home & away)
   (m "Brasileirão Série A" 2019 "1" "2019-04-28" "Flamengo" "Palmeiras" 3 1)
   (m "Brasileirão Série A" 2019 "2" "2019-05-05" "Palmeiras" "Flamengo" 0 2)
   (m "Brasileirão Série A" 2019 "3" "2019-05-12" "Flamengo" "Santos" 4 0)
   (m "Brasileirão Série A" 2019 "4" "2019-05-19" "Santos" "Flamengo" 1 1)
   (m "Brasileirão Série A" 2019 "5" "2019-05-26" "Palmeiras" "Santos" 2 2)
   (m "Brasileirão Série A" 2019 "6" "2019-06-02" "Santos" "Palmeiras" 0 1)
   ;; A different season for the same teams
   (m "Brasileirão Série A" 2018 "1" "2018-04-28" "Flamengo" "Palmeiras" 0 1)
   ;; Fla-Flu derby across competitions
   (m "Brasileirão Série A" 2019 "7" "2019-09-03" "Flamengo" "Fluminense" 2 1)
   (m "Copa do Brasil"      2019 "Quarterfinals" "2019-07-10" "Fluminense" "Flamengo" 1 0)
   ;; Libertadores blowout for biggest-wins
   (m "Copa Libertadores"   2012 nil "2012-05-27" "Santos" "Bolivar" 8 0)])

(def players
  [{:id 1 :name "Neymar Jr" :age 27 :nationality "Brazil" :overall 92 :potential 92
    :club "Paris Saint-Germain" :position "LW" :jersey 10 :skills {}}
   {:id 2 :name "Gabriel Barbosa" :age 22 :nationality "Brazil" :overall 80 :potential 86
    :club "Flamengo" :position "ST" :jersey 9 :skills {}}
   {:id 3 :name "Bruno Henrique" :age 28 :nationality "Brazil" :overall 78 :potential 78
    :club "Flamengo" :position "LW" :jersey 27 :skills {}}
   {:id 4 :name "Diego Alves" :age 33 :nationality "Brazil" :overall 79 :potential 79
    :club "Flamengo" :position "GK" :jersey 1 :skills {}}
   {:id 5 :name "L. Messi" :age 31 :nationality "Argentina" :overall 94 :potential 94
    :club "FC Barcelona" :position "RF" :jersey 10 :skills {}}])

(def db {:matches matches :players players})
